New feature - 6: 1) посчитать все доставленные ордера для каждого юзе…

scripa01 opened the pull request:
…ра (userID,user fullname, count, id ордеров через запятую )

                2)  посчитать все недоставленные  ордера для каждого юзера (userID, user fullname, count, id ордеров через запятую )
                3) Найти текст ревью и имя + фамилию подльзователя который его написал (userId, userFullName, reviewText)

scripa01 committed:
New feature - 6: 1) посчитать все доставленные ордера для каждого юзера (userID,user fullname, count, id ордеров через запятую )
                2)  посчитать все недоставленные  ордера для каждого юзера (userID, user fullname, count, id ордеров через запятую )
                3) Найти текст ревью и имя + фамилию подльзователя который его написал (userId, userFullName, reviewText)

ngherta commented on src/main/java/com/example/springexploring/controller/OrderStatisticController.java:
```diff
@@ -26,4 +27,14 @@ public ResponseEntity<List<OrderStatisticDTO>> findAll() {
+
+    @GetMapping("/count/non-delivered")
+    public ResponseEntity<List<CountOrdersByUserDTO>> countNonDeliveredOrderByUser() {
+        return ResponseEntity.ok(orderStatisticService.getCountOfOrdersByStatus("CREATED"));
```
bad, you have ENUM

ngherta commented on src/main/java/com/example/springexploring/controller/OrderStatisticController.java:
```diff
@@ -26,4 +27,14 @@ public ResponseEntity<List<OrderStatisticDTO>> findAll() {
+
+    @GetMapping("/count/delivered")
+    public ResponseEntity<List<CountOrdersByUserDTO>> countDeliveredOrderByUser() {
+        return ResponseEntity.ok(orderStatisticService.getCountOfOrdersByStatus("DELIVERED"));
```
not good 
you have ENUM

ngherta commented on src/main/java/com/example/springexploring/dto/CountOrdersByUserDTO.java:
```diff
@@ -0,0 +1,16 @@
+
+@Data
+@AllArgsConstructor
+public class CountOrdersByUserDTO {
```
Why are fields default and not private?

ngherta commented on src/main/java/com/example/springexploring/dto/FindTextReviewAndNameOfUserWhoReviewedDTO.java:
```diff
@@ -0,0 +1,14 @@
+
+@Data
+@AllArgsConstructor
+public class FindTextReviewAndNameOfUserWhoReviewedDTO {
```
Why are fields default and not private?

ngherta commented on src/main/java/com/example/springexploring/service/OrderStatisticServiceImpl.java:
```diff
@@ -30,4 +35,17 @@ public List<OrderStatisticDTO> findAll() {
     }
+
+    @Override
+    public List<CountOrdersByUserDTO> getCountOfOrdersByStatus(String status) {
```
Use enum insead of simple string

ngherta commented on src/main/java/com/example/springexploring/service/OrderStatisticServiceImpl.java:
```diff
@@ -30,4 +35,17 @@ public List<OrderStatisticDTO> findAll() {
+    @Override
+    public List<CountOrdersByUserDTO> getCountOfOrdersByStatus(String status) {
+        List<CountOrdersByUserProjections> countListProjections = orderRepository.countOrdersByStatusAndUserWhoOrd(status);
+        List<CountOrdersByUserDTO> countByStatusAndUser = new ArrayList<>();
```
You have mapper.mapList() method

scripa01 committed:
New feature - 6: 1) посчитать все доставленные ордера для каждого юзера (userID,user fullname, count, id ордеров через запятую )
                2)  посчитать все недоставленные  ордера для каждого юзера (userID, user fullname, count, id ордеров через запятую )
                3) Найти текст ревью и имя + фамилию подльзователя который его написал (userId, userFullName, reviewText)
